mailgun/route_handlers.py

```diff
@@ -141,11 +141,22 @@ def _handle_recipient(request, recipient):
             u'course id %s, so we cannot prepend a short title to the '
             u'email subject, or check the super senders.', ml.canvas_course_id)
 
-    # get the list of staff and member addresses.  always include staff addresses
-    # in the members list, so they can send to any list in the course, and receive
-    # all mail for the course.
+    member_addresses = set([m['address'].lower() for m in ml.members])
+
+    # conditionally include staff addresses in the members list. If always_mail_staff is true
+    # all staff will receive the email
     teaching_staff_addresses = ml.teaching_staff_addresses
-    member_addresses = teaching_staff_addresses.union(
+    if ml.course_settings and ml.course_settings.always_mail_staff:
+        member_addresses = member_addresses.union(teaching_staff_addresses)
+    elif ml.course_settings == None:
+        member_addresses = member_addresses.union(teaching_staff_addresses)
+        # if the course settings object does not exist create it, this equates to True
```

Comment:
It looks like this code makes assumptions about the default value for the model field `always_mail_staff`. It's probably better to just rely on that value entirely, so that any change to the model will automatically be handled here without having to remember that there's a dependency.

So perhaps something like:

```
# if the course settings object does not exist, create it to initialize the defaults
if ml.course_settings == None:
    ml.course_settings = CourseSettings.objects.create(canvas_course_id=ml.canvas_course_id)
    ml.save()

if ml.course_settings and ml.course_settings.always_mail_staff:
    member_addresses = member_addresses.union(teaching_staff_addresses)
```

This is also a little DRYer.